x-pack/plugins/security_solution/common/detection_engine/prebuilt_rules/rule_field_updates_and_conflicts.md

```diff
@@ -0,0 +1,604 @@
+**Epic:** https://github.com/elastic/security-team/issues/1974 (internal)
+
+## Summary
+
+This "PR" provides some examples we could use as a foundation for discussing and designing the UI for showing rule field diffs and resolving conflicts.
+
+First, it describes six typical situations we can have for any rule field.
+
+Then, it expands on how each of these six situations can look like for the following fields:
+
+- rule name
+- rule tags
+- data source: index patterns or data view
+- rule query
+
+NOTE: You can comment this PR description line-by-line in `security_solution/common/detection_engine/prebuilt_rules/rule_field_updates_and_conflicts.md`.
+
+There's no intention to merge this PR.
+
+## Typical situations
+
+Given a rule that can be upgraded, for every top-level field of this rule, we will have 3 versions of the field:
+
+- **Base version**: the original stock version of the field shipped by Elastic and installed by the user.
+- **Current version**: the version currently installed, potentially with user’s customizations.
+  - If the user didn't customize this field, current == base.
+  - If the field was customized, current != base.
+- **Target version**: the latest stock version of the field shipped by Elastic and available for upgrade.
+
+Using the above 3 versions, the app will attempt to automatically merge them into the final **Merged version**. The merge can either succeed, or fail with a **conflict**. Users will have to resolve conflicts manually in the UI. This automatic merge is supposed to improve the UX by reducing the amount of work our users will have to do during rule upgrade.
+
+Here are six typical situations that we will have to handle in the app:
+
+1. **`base=A, current=A, target=A => merged=A, conflict=false`**
+    - **Situation**: Stock rule was installed. The user didn't customize the field. Elastic doesn't have any updates for this field.
+    - **How to handle**: ❓ Don't show the field in the upgrade flyout?
+2. **`base=A, current=A, target=B => merged=B, conflict=false`**
+    - **Situation**: Stock rule was installed. The user didn't customize the field. Elastic updated this field in the latest version.
+    - **How to handle**: Show this field in the flyout. Automatically pick the updated version from Elastic as the merged one.
+3. **`base=A, current=B, target=A => merged=B, conflict=false`**
+    - **Situation**: Stock rule was installed. The user customized the field. Elastic doesn't have any updates for this field.
+    - **How to handle**: ❓ Don't show the field in the upgrade flyout? Or show the field, and automatically pick the current customized version as the merged one?
+4. **`base=A, current=B, target=B => merged=B, conflict=false`**
+    - **Situation**: Stock rule was installed. The user customized the field. Elastic updated this field in the latest version to exactly the same value as the user did it in their cluster.
+    - **How to handle**: ❓ Don't show the field in the upgrade flyout?
```

Comment:
Similar to situation 1, there ends up being no delta would this rule even show up in the `Rule Updates` table?

Reply:
I explained the difference in https://github.com/elastic/kibana/pull/148913#discussion_r1081389335